resources/services/iam/roles_access_advisor_details.go

```diff
@@ -0,0 +1,184 @@
+package iam
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/cloudquery/cq-provider-aws/client"
+	"github.com/cloudquery/cq-provider-sdk/provider/diag"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+//go:generate cq-gen --resource roles_access_advisor_details --config gen.hcl --output .
+func RolesAccessAdvisorDetails() *schema.Table {
+	return &schema.Table{
+		Name:          "aws_iam_access_advisor_details",
+		Description:   "IAM Access Advisor details of the IAM resources (users, groups, roles, policies)",
+		Resolver:      fetchIamRolesAccessAdvisorDetails,
+		Multiplex:     client.AccountMultiplex,
+		IgnoreError:   client.IgnoreAccessDeniedServiceDisabled,
+		DeleteFilter:  AccessAdvisorFilter(ROLE),
+		IgnoreInTests: true,
+		Options:       schema.TableCreationOptions{PrimaryKeys: []string{"account_id", "service_namespace"}},
+		Columns: []schema.Column{
+			{
+				Name:        "account_id",
+				Description: "The AWS Account ID of the resource.",
```

Comment:
Rather than duplicating all of the column definitions in each resource can we create a single place for them and then just reference it? This way it will be easier for people to understand that it is a single table and if they make changes in one file it will impact multiple resources